apps/src/lib/node/ledger/shell/finalize_block.rs

```diff
@@ -2234,6 +2235,10 @@ mod test_finalize_block {
             })
             .expect("Test failed")[0];
 
+        // the merkle tree root should not change after finalize_block
+        let root_post = shell.shell.wl_storage.storage.block.tree.root();
+        assert_eq!(root_pre.0, root_post.0);
```

Comment:
this one also seems to pass without the fixes

Reply:
this one was already passing before because `TxType::Decrypted` didn't commit anything directly to storage so this is ok